src/configprovider-lldb.ts

```diff
@@ -0,0 +1,65 @@
+import * as vscode from "vscode";
+import * as path from "path";
+import { getMesonTargets } from "./introspection";
+import { Target } from "./types";
+import { extensionConfiguration, getTargetName } from "./utils";
+
+export class DebugConfigurationProviderLldb implements vscode.DebugConfigurationProvider {
+  private path: string;
+
+  constructor(path: string) {
+    this.path = path;
+  }
+
+  async createDebugConfiguration(target: Target): Promise<vscode.DebugConfiguration> {
+    const targetName = await getTargetName(target);
+    return {
+      type: "lldb",
+      name: `Debug ${target.name} (lldb)`,
+      request: "launch",
+      cwd: path.dirname(this.path),
+      program: target.filename[0],
+      args: [],
+      preLaunchTask: `Meson: Build ${targetName}`,
+    };
+  }
+
+  async provideDebugConfigurations(
+    folder: vscode.WorkspaceFolder | undefined,
+    token?: vscode.CancellationToken,
+  ): Promise<vscode.DebugConfiguration[]> {
+    let targets = await getMesonTargets(this.path);
+
+    let configDebugOptions = extensionConfiguration("debugOptions");
+
+    const executables = targets.filter((target) => target.type == "executable");
+    let ret: vscode.DebugConfiguration[] = [];
+
+    for (const target of executables) {
+      if (!target.target_sources?.some((source) => ["cpp", "c"].includes(source.language))) {
+        continue;
+      }
+
+      let debugConfiguration = await this.createDebugConfiguration(target);
+      ret.push({ ...configDebugOptions, ...debugConfiguration });
+    }
```

Comment:
Looks like some of these lets can be const to me.

Reply:
This was in the code previously. I fixed it in e817bb1.